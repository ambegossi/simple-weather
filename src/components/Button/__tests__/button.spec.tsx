import { render } from '@testing-library/react-native';
import { ThemeProvider } from 'styled-components/native';

import { Button } from '..';

import theme from '../../../styles/themes/light';

const Providers: React.FC = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

describe('Button component', () => {
  it('should has a title', () => {
    const { getByText } = render(<Button title="Button title" />, {
      wrapper: Providers,
    });

    getByText('Button title');
  });
});
