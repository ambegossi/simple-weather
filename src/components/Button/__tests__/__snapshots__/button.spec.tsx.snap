// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#1A1A1A",
      "borderRadius": 8,
      "justifyContent": "center",
      "marginBottom": 8,
      "opacity": 1,
      "paddingBottom": 15,
      "paddingLeft": 15,
      "paddingRight": 15,
      "paddingTop": 15,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 15,
          "fontWeight": "bold",
        },
      ]
    }
  >
    Button title
  </Text>
</View>
`;
